vaas/vaas/metrics/prometheus.py

```diff
@@ -0,0 +1,45 @@
+from datetime import timedelta
+from typing import Optional, TypedDict, Union
+from django.conf import settings
+from prometheus_client import CollectorRegistry, Gauge, Summary, push_to_gateway
+
+from vaas.metrics.models import Metrics
+
+
+class MetricsBucket(TypedDict, total=False):
+    name: str
+    metric: Union[Summary, Gauge]
+
+
+class PrometheusClient:
+    def __init__(self) -> None:
+        self.host: str = f'{settings.PROMETHEUS_GATEWAY_HOST}:{settings.PROMETHEUS_GATEWAY_PORT}'
+        self.job: str = 'vaas'
+        self.registry: CollectorRegistry = CollectorRegistry()
+
+    def push(self) -> None:
+        push_to_gateway(gateway=self.host, job=self.job, registry=self.registry)
+
+
+class PrometheusMetrics(Metrics):
+    def __init__(self) -> None:
+        super().__init__(
+            client=PrometheusClient())
+        self.metrics_bucket: MetricsBucket = {}
+
+    def _get_or_create(self, name: str, type: Union[Summary, Gauge]) -> Optional[Union[Summary, Gauge]]:
+        name_with_suffix: str = f'{name}_microseconds'
+        if name:
```

Comment:
It is redundant, you have already checked if the name is not empty

Reply:
Right.